### DISCUSSION TITLE
Instantiating bar lines??

### DISCUSSION THREAD

**Discussion by cad12345** [melanchall/drywetmidi#202] [Q&A] [answered]
Hi,
Just a newbie here. I’m building a side scrolling music score in Unity using drywetmidi. I have most things working, expect for instantiating bar lines. 
I would like to instantiate bar line game objects that will scroll across the screen in the correct position like the notes. 
I can’t work out the correct timing of when they should be instantiated. 

Thanks!

**Comment by melanchall** [comment, marked as answer]
Hi,

Can you please clarify what you exactly need? You want to get times in milliseconds (or in another format) of all bar lines in a MIDI file?

Use this code please:

```csharp
var barBeatTicksDuration = midiFile.GetDuration<BarBeatTicksTimeSpan>();
var bars = barBeatTicksDuration.Bars;

var tempoMap = midiFile.GetTempoMap();
var barLinesMilliseconds = Enumerable
    .Range(0, (int)bars + 1)
    .Select(bar => TimeConverter
        .ConvertTo<MetricTimeSpan>(new BarBeatTicksTimeSpan(bar), tempoMap)
        .TotalMilliseconds)
    .ToArray();
```

**Reply by cad12345** [reply]
Thanks for responding. Yes, I would like to get all of the bar lines of a midi file in milliseconds. 
With that information, I can then instantiate bar lines based on those time stamps.

**Reply by melanchall** [reply]
I've updated my answer. Please see the code there.

**Reply by cad12345** [reply]
That worked. Thank you so much for your time!
